### PR TITLE
Add option to choose between track or album normalisation gain

### PR DESCRIPTION
**Pull request by sashahilton00**
Add option to choose between track or album normalisation gain, set default normalisation gain type to album. If there's something I've missed here, let me know.

### CODE REVIEW

**Comment by sashahilton00** [comment]
Relevant to the conversation in #162 and closes #572

**Comment by sashahilton00** [comment]
@philippe44 if you wouldn't mind testing this, using the `--enable-volume normalisation --normalisation-gain-type track` and `--enable-volume normalisation --normalisation-gain-type album` options, that would be great, since I'm not going to be able to hear the difference through earpods to verify that it works as expected.

**Comment by philippe44** [comment]
I'm happy to help, but are you sure this message is for me? My PR does not touch/impact volume&gain normalization as it's a passthrough

**Comment by sashahilton00** [comment]
Oops, pinged the wrong person. @slartibartfast3 see above.

**Comment by slartibartfast3** [comment]
> Oops, pinged the wrong person. @slartibartfast3 see above.

I can definitely test of you tell me what I need to do. I am using Michael Herger's plugin for Squeezebox. You will easily hear the difference if you listen to the last track on Pink Floyd "The Wall". It is meant to be very low volume so using track gain is obviously far louder.

**Comment by slartibartfast3** [comment]
> 
> 
> Oops, pinged the wrong person. @slartibartfast3 see above.

Sorry it took so long but I finally managed to install everything and I can confirm it works. I used the transition between the final two tracks on "The Wall" to test it and with track gain there is a massive jump in volume whereas with album gain the tracks flow together perfectly.

**Comment by sashahilton00** [comment]
Unless there are any objections, I'll add the documentation to the wiki and merge this in a few days.

**Review by herrernst — CHANGES_REQUESTED** [review]
Looks good to me, thank you! Only one small issue in arguments doc, see comment.

**Comment by sashahilton00** [comment]
Yes you're right. I amended the default at the last minute and forgot to update the doc, will change that later

**Comment by slartibartfast3** [comment]
Is the label "breaking change" as serious as it sounds?

**Comment by ashthespy** [comment]
> 
> 
> Is the label "breaking change" as serious as it sounds?

It depends ;-)
image

https://xkcd.com/1172/

**Comment by sashahilton00** [comment]
I labelled as breaking mainly so that I have an easy reference point as to what needs to be updated in the wiki, and what needs to go into the changelog for next release.

File: src/main.rs [hunk 1]
```diff
@@ -177,6 +177,12 @@ fn setup(args: &[String]) -> Setup {
             "enable-volume-normalisation",
             "Play all tracks at the same volume",
         )
+        .optopt(
+            "",
+            "normalisation-gain-type",
+            "Specify the normalisation gain type to use - [track, album]. Default is track.",
```
[the remark this answers is not in the document]
Comment (sashahilton00) [reply src/main.rs]: fixed.

**Pull request merged**